pkg/conversion/scheme.go

```diff
@@ -388,13 +399,13 @@ func (s *Scheme) ObjectVersionAndKind(obj interface{}) (apiVersion, kind string,
 		return "", "", err
 	}
 	t := v.Type()
-	version, vOK := s.typeToVersion[t]
-	kinds, kOK := s.typeToKind[t]
-	if !vOK || !kOK {
+
+	gvks, ok := s.typeToGVK[t]
+	if !ok {
 		return "", "", &notRegisteredErr{t: t}
 	}
-	apiVersion = version
-	kind = kinds[0]
+	apiVersion = gvks[0].GroupVersion().String()
```

Comment:
> Should this method signature be changed to return a GroupVersion instead of a string?

Yes, but I have to limit ripples.  That forces updates in another 300 spots.